strawberry/type.py

```diff
@@ -51,18 +53,25 @@ class StrawberryContainer(StrawberryType):
     def __init__(self, of_type: Union[StrawberryType, type]):
         self.of_type = of_type
 
-    def __hash__(self) -> int:
-        return hash((self.__class__, self.of_type))
-
     def __eq__(self, other: object) -> bool:
         if isinstance(other, StrawberryType):
-            if isinstance(other, StrawberryContainer):
+            if type(self) == type(other):
+                other = cast(StrawberryContainer, other)
```

Comment:
Not sure I understand why two objects being of the same type means that one of them is a container type. Can you elaborate here?